runtime/common/Future.cpp

```diff
@@ -40,9 +40,28 @@ sample_result future::get() {
       resultResponse = client.get(jobGetPath, "", headers, false,
                                   serverHelper->getCookies());
     }
-    auto c = serverHelper->processResults(resultResponse, id.first);
 
-    if (isObserve) {
+    if (resultType == ExecutionContextType::run) {
+      QirServerHelper *qirServerHelper =
+          dynamic_cast<QirServerHelper *>(serverHelper.get());
+      if (!qirServerHelper)
+        throw std::runtime_error(
+            "cudaq::details::future::getRawOutput() requires QIR Server Helper "
+            "but QPU named " +
+            qpuName + " is not implementing it.");
```

Comment:
Can we make this message clearer and actionable, something like - 
```suggestion
        throw std::runtime_error("To support `run` API, " + qpuName + "must inherit `QirServerHelper` class");
```